### PR TITLE
[expo-analytics-amplitude][android] rewriting java source code to Kotlin

### PR DESCRIPTION
**Pull request by kkafar**
# Why

This task was done as a part of migrating expo modules to Kotlin project.

# How

The code had been initially converted from Java to Kotlin with usage of Android Studio's conversion tool. Then it was manually adjusted / improved.

# Test Plan

- - - AFAIK there are no tests for this module :/

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).

cc @lukmccall

### CODE REVIEW

File: packages/expo-analytics-amplitude/android/src/main/java/expo/modules/analytics/amplitude/AmplitudePackage.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+import android.content.Context
+import org.unimodules.core.BasePackage
+import org.unimodules.core.ExportedModule
+
+class AmplitudePackage : BasePackage() {
+  override fun createExportedModules(context: Context): List<ExportedModule> {
+    return listOf(AmplitudeModule(context) as ExportedModule)
+  }
+}
```
Comment (lukmccall) [inline packages/expo-analytics-amplitude/android/src/main/java/expo/modules/analytics/amplitude/AmplitudePackage.kt:11]:
```suggestion
}

```

File: packages/expo-analytics-amplitude/android/src/main/java/expo/modules/analytics/amplitude/AmplitudePackage.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package expo.modules.analytics.amplitude
+
+import android.content.Context
+import org.unimodules.core.BasePackage
+import org.unimodules.core.ExportedModule
+
+class AmplitudePackage : BasePackage() {
+  override fun createExportedModules(context: Context): List<ExportedModule> {
+    return listOf(AmplitudeModule(context) as ExportedModule)
```
Comment (lukmccall) [inline packages/expo-analytics-amplitude/android/src/main/java/expo/modules/analytics/amplitude/AmplitudePackage.kt:9]:
```suggestion
    return listOf(AmplitudeModule(context))
```

File: packages/expo-analytics-amplitude/android/src/main/java/expo/modules/analytics/amplitude/AmplitudeModule.kt [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+      trackingOptions.disableVersionName()
+    }
+    if (mClient != null) {
+      mClient!!.setTrackingOptions(trackingOptions)
+    } else {
+      mPendingTrackingOptions = trackingOptions
+    }
+    promise.resolve(null)
+  }
+}
```
Comment (lukmccall) [inline packages/expo-analytics-amplitude/android/src/main/java/expo/modules/analytics/amplitude/AmplitudeModule.kt:146]:
```suggestion
}

```

File: packages/expo-analytics-amplitude/android/src/main/java/expo/modules/analytics/amplitude/AmplitudeModule.kt [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  fun initializeAsync(apiKey: String, promise: Promise) {
+    mClient = getClient(apiKey)
+    val client = mClient
+    if (mPendingTrackingOptions != null) {
+      client?.setTrackingOptions(mPendingTrackingOptions)
+    }
+    client?.initialize(context, apiKey)
+    promise.resolve(null)
+  }
+
```
Comment (lukmccall) [inline packages/expo-analytics-amplitude/android/src/main/java/expo/modules/analytics/amplitude/AmplitudeModule.kt:38]:
```suggestion
```
🧹

**Review by lukmccall — COMMENTED** [review]
Looks good to me 💪 Very solid work. However, we should test it before landing it on master 😉

**Review by tsapeta — APPROVED** [review]

**Pull request merged**